Essentials/src/main/java/com/earth2me/essentials/Settings.java

```diff
@@ -1703,13 +1703,21 @@ public boolean isConfirmCommandEnabledByDefault(final String commandName) {
         return getDefaultEnabledConfirmCommands().contains(commandName.toLowerCase());
     }
 
-    private boolean _isTeleportBackWhenFreedFromJail() {
-        return config.getBoolean("teleport-back-when-freed-from-jail", true);
+    private TeleportWhenFreePolicy _getTeleportWhenFreePolicy() {
+        if (config.hasProperty("teleport-back-when-freed-from-jail")) {
+            return config.getBoolean("teleport-back-when-freed-from-jail", true) ? TeleportWhenFreePolicy.BACK : TeleportWhenFreePolicy.OFF;
+        }
+
+        try {
+            return TeleportWhenFreePolicy.valueOf(config.getString("teleport-when-freed", "back"));
+        } catch (IllegalArgumentException e) {
+            return TeleportWhenFreePolicy.BACK;
+        }
```

Comment:
```suggestion
if (config.hasProperty("teleport-when-freed")) {
            // snakeyaml more like cursedyaml
            final String value = config.getString("teleport-when-freed", "back").replace("false", "off");
            try {
                return TeleportWhenFreePolicy.valueOf(value.toUpperCase(Locale.ROOT));
            } catch (IllegalArgumentException e) {
                throw new RuntimeException("Invalid value \"" + value + "\" for config option \"teleport-when-freed\"!", e);
            }
        }

        return TeleportWhenFreePolicy.BACK;
```

- `TeleportWhenFreePolicy.valueOf(config.getString("teleport-when-freed", "back"))` doesn't allow for non-uppercase values
- If `teleport-when-freed` exists but has an invalid value, we should error instead of defaulting
- Is it worth adding a `InvalidConfigValueException extends RuntimeException` for this? There's possible other cases where we should throw for invalid settings so the user knows what's wrong, instead of silently defaulting and leaving users wondering why settings aren't working